R/joyn-merge.R

```diff
@@ -212,12 +212,8 @@ joyn <- function(x,
   #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   start_joyn <- Sys.time()
   # copy objects if data.tables
-  if (any(class(x) == "data.table")) {
-    x <- copy(x)
-  }
-  if (any(class(y) == "data.table")) {
-    y <- copy(y)
-  }
+  x <- copy(x)
+  y <- copy(y)
```

Comment:
Why did you remove the condition? If the data is big, this `copy()` will slow down the function. You can use `inherits(x, "data.table")`.

Reply:
When using check_by_vars() it was modifying the names of the input tables by reference, so that the column names in x and y had been changed to `keyby1`. Which is why I used `copy()` internally. I can look for an alternative solution that is more memory efficient for large data frames

Reply:
GOt it. I think we can proceed for now, but that `data.table` code should be avoided. Thanks.